pco_cli/generate_randoms.py

```diff
@@ -1,203 +1,296 @@
 # python 3
 # pip requirement: numpy, pyarrow
 
+import argparse
 import numpy as np
 from datetime import datetime
 import os
+from pathlib import Path
 
-n = 1000000
+n = 1_000_000
 max_n = 2 ** 24 - 1
 
-base_dir = 'data'
-
-os.makedirs(f'{base_dir}/txt', exist_ok=True)
-os.makedirs(f'{base_dir}/binary', exist_ok=True)
-
-def write_generic(strs, arr, full_name):
+def write_generic(strs, arr, full_name, base_dir):
   print(f'writing {full_name}...')
   joined = '\n'.join(strs)
-  with open(f'{base_dir}/txt/{full_name}.txt', 'w') as f:
+  with open(base_dir / 'txt' / f'{full_name}.txt', 'w') as f:
     f.write(joined)
-  with open(f'{base_dir}/binary/{full_name}.bin', 'wb') as f:
+  with open(base_dir / 'binary' / f'{full_name}.bin', 'wb') as f:
     f.write(arr.tobytes())
 
-def write_i32(arr, name):
+WRITERS = {}
+def writer(f):
+  name = f.__name__
+  assert name.startswith('write_')
+  dtype = name.removeprefix('write_')
+  WRITERS[dtype] = f
+  return f
+
+@writer
+def write_i32(arr, name, base_dir):
   if arr.dtype != np.int32:
     arr = np.floor(arr).astype(np.int32)
   strs = [str(x) for x in arr]
   full_name = f'i32_{name}'
-  write_generic(strs, arr, full_name)
+  write_generic(strs, arr, full_name, base_dir)
 
-def write_u32(arr, name):
+@writer
+def write_u32(arr, name, base_dir):
   if arr.dtype != np.uint32:
     arr = np.floor(arr).astype(np.uint32)
   strs = [str(x) for x in arr]
   full_name = f'u32_{name}'
-  write_generic(strs, arr, full_name)
+  write_generic(strs, arr, full_name, base_dir)
 
-def write_i64(arr, name):
+@writer
+def write_i64(arr, name, base_dir):
   if arr.dtype != np.int64:
     arr = np.floor(arr).astype(np.int64)
   strs = [str(x) for x in arr]
   full_name = f'i64_{name}'
-  write_generic(strs, arr, full_name)
+  write_generic(strs, arr, full_name, base_dir)
 
-def write_bool(arr, name):
```

Comment:
I think we can just delete this one now - it's unused

Reply:
Done.